### PR TITLE
Support Ruby 3.2

### PR DESCRIPTION
**Pull request by sgerrand**
💁 The implementation of `Refinement#include` was removed in Ruby version 3.2, which results in errors like the following being returned:

	TypeError: Refinement#include has been removed

This change updates the implementation to use `#extend` instead as a way of maintaining interface parity and enabling this gem to be used with modern Rubies.

### CODE REVIEW

**Comment by sgerrand** [comment]
👋 @palkan @ssnickolay: Are you able to approve the pending workflows and review these changes?

File: lib/clowne/ext/yield_self_then.rb [hunk 1]
```diff
@@ -15,10 +15,10 @@ def yield_self
       end
 
       # See https://github.com/jruby/jruby/issues/5220
-      ::Object.include(Ext) if RUBY_PLATFORM.match?(/java/i)
+      ::Object.extend(Ext) if RUBY_PLATFORM.match?(/java/i)
```
[the remark this answers is not in the document]
Comment (sgerrand) [reply lib/clowne/ext/yield_self_then.rb:18]: Yes, I'm happy to do that.

[the remark this answers is not in the document]
Comment (ssnickolay) [reply lib/clowne/ext/yield_self_then.rb:18]: So we agreed to remove this refinement in the current PR, right?

[the remark this answers is not in the document]
Comment (palkan) [reply lib/clowne/ext/yield_self_then.rb:18]: Yeah, we no longer need it.

[the remark this answers is not in the document]
Comment (sgerrand) [reply lib/clowne/ext/yield_self_then.rb]: Applied in e6efb3a.

File: lib/clowne/ext/yield_self_then.rb [hunk 2]
```diff
@@ -14,11 +14,8 @@ def yield_self
         alias_method :then, :yield_self
       end
 
-      # See https://github.com/jruby/jruby/issues/5220
-      ::Object.include(Ext) if RUBY_PLATFORM.match?(/java/i)
-
       refine Object do
-        include Ext
+        extend Ext
```
Comment (palkan) [inline lib/clowne/ext/yield_self_then.rb:18]: We should remove this module (and file) completely; and all the references. `Object#then` is available on all supported platforms/versions.

Comment (sgerrand) [reply lib/clowne/ext/yield_self_then.rb:18]: Thanks for clarifying that point, would have been useful to have had that guidance earlier but we are where we are.

**Comment by sgerrand** [comment]
still waiting for an approval

**Comment by palkan** [comment]
Thanks! Released in 1.5.0

**Pull request merged**